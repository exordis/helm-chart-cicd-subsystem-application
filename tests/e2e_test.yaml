suite: Ensure no changes in manifests compared to saved snapshot (no unintentional changes)
templates:
  - templates/*.yaml
values:
  - ../values.linter.yaml
tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}
