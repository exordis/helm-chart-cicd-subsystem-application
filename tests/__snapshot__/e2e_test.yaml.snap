manifest should match snapshot:
  1: |
    apiVersion: v1
    data:
      value1: test
      value2: "123"
    kind: ConfigMap
    metadata:
      annotations:
        extra-annotation: annotation value
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        extra-label: label value
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-config-map-full-metadata
      namespace: my-namespace
  2: |
    apiVersion: v1
    data:
      value: test1
    kind: ConfigMap
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-config-map-with-containers
      namespace: cicd-test
  3: |
    apiVersion: v1
    data:
      value: test2
    kind: ConfigMap
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-config-map-without-containers
      namespace: cicd-test
  4: |
    apiVersion: v1
    data:
      EXORDIS_APPLICATION: cicd-sample
      EXORDIS_ENVIRONMENT: test
      EXORDIS_INSTANCE: docs
      EXORDIS_PRODUCT: Some Product
      EXORDIS_SUBSYSTEM: cicd
      SUBSYSTEM: wrong-subsystem
      TEST: 123
    kind: ConfigMap
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-envs
      namespace: cicd-test
  5: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-cleanup
      namespace: cicd-test
    spec:
      concurrencyPolicy: Forbid
      failedJobsHistoryLimit: 1
      jobTemplate:
        spec:
          template:
            metadata:
              annotations: null
              labels:
                app.kubernetes.io/component: cicd-sample
                app.kubernetes.io/instance: docs
                app.kubernetes.io/managed-by: helm
                app.kubernetes.io/name: cicd-sample
                app.kubernetes.io/part-of: cicd
                app.kubernetes.io/version: 1.0.0
                exordis/application: cicd-sample
                exordis/application-instance: docs
                exordis/application-type: service
                exordis/environment: test
                exordis/product: Some Product
                exordis/subsystem: cicd
                helm.sh/chart: cicd-subsystem-application-0.1.0
            spec:
              containers:
                - args:
                    - tag
                    - delete
                    - --registry
                  envFrom:
                    - configMapRef:
                        name: cicd-sample-docs-config-map-without-containers
                    - configMapRef:
                        name: cicd-sample-docs-envs
                    - secretRef:
                        name: cicd-sample-docs-gitlab-token
                    - secretRef:
                        name: cicd-sample-docs-secret-with-containers
                    - secretRef:
                        name: cicd-sample-docs-secret-without-containers
                  image: registry.gitlab.com/knst/cy/registry/cicd-docker-registry-cleaner:0.1.0
                  imagePullPolicy: IfNotPresent
                  name: main
                  resources:
                    limits:
                      cpu: 200m
                      memory: 1024Mi
                    requests:
                      cpu: 2m
                      memory: 70Mi
                  volumeMounts:
                    - mountPath: /some-data
                      name: volume2
              initContainers:
                - envFrom:
                    - configMapRef:
                        name: cicd-sample-docs-config-map-without-containers
                    - configMapRef:
                        name: cicd-sample-docs-envs
                    - secretRef:
                        name: cicd-sample-docs-gitlab-token
                    - secretRef:
                        name: cicd-sample-docs-secret-without-containers
                  image: registry.gitlab.com/cleanup-init:1.0.0
                  imagePullPolicy: IfNotPresent
                  name: cleanup-init
                  resources:
                    limits:
                      cpu: 200m
                      memory: 1024Mi
                    requests:
                      cpu: 2m
                      memory: 70Mi
                  volumeMounts: null
                - envFrom:
                    - configMapRef:
                        name: cicd-sample-docs-config-map-without-containers
                    - configMapRef:
                        name: cicd-sample-docs-envs
                    - secretRef:
                        name: cicd-sample-docs-gitlab-token
                    - secretRef:
                        name: cicd-sample-docs-secret-without-containers
                  image: registry.gitlab.com/software-development/subsystems/some-subsystem/data-content-templates:1.24.0-dev.5
                  imagePullPolicy: Always
                  name: templates
                  resources:
                    limits:
                      cpu: 200m
                      memory: 1024Mi
                    requests:
                      cpu: 2m
                      memory: 70Mi
                  volumeMounts:
                    - mountPath: /data/templates/generated
                      name: templates-data-templates-generated
                    - mountPath: /data/templates/original
                      name: templates-data-templates-original
              restartPolicy: Never
              serviceAccountName: default
              volumes:
                - name: mypvc
                  persistentVolumeClaim:
                    claimName: cicd-sample-docs-mypvc
                - emptyDir:
                    sizeLimit: 200Mi
                  name: templates-data-templates-generated
                - emptyDir:
                    sizeLimit: 200Mi
                  name: templates-data-templates-original
                - emptyDir:
                    sizeLimit: 100Mi
                  name: volume2
          ttlSecondsAfterFinished: 86400
      schedule: 0 6 * * *
  6: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs
      namespace: cicd-test
    spec:
      replicas: 3
      revisionHistoryLimit: 1
      selector:
        matchLabels:
          exordis/application: cicd-sample
          exordis/application-instance: docs
          exordis/environment: test
          exordis/subsystem: cicd
      strategy:
        rollingUpdate:
          maxSurge: 25%
          maxUnavailable: 25%
        type: RollingUpdate
      template:
        metadata:
          annotations:
            checksum/configMap.config-map-with-containers: 48b7e320feb03b105aa89ed415344c030af90d34a34c66f56146d9f0fd2da3e3
            checksum/configMap.config-map-without-containers: 1dbfa263d171e565d412b904c69bdf60fc6bc8db2fcb31e13049242a8bbdecac
            checksum/configMap.envs: 7804c1a2bc6fb98bc7b87c036f6bdb678f897e534204c2e4efd7d881bf5b2924
            checksum/secret.secret-with-containers: 8e0836f50407f40dc36bf200910416c00f82b8bf5284ab96e908b1d709eb65e3
            checksum/secret.secret-without-containers: 2e931d517cebd760f5460ea95663a82d8df2673e2c075918c21b33c96bfd6686
          labels:
            app.kubernetes.io/component: cicd-sample
            app.kubernetes.io/instance: docs
            app.kubernetes.io/managed-by: helm
            app.kubernetes.io/name: cicd-sample
            app.kubernetes.io/part-of: cicd
            app.kubernetes.io/version: 1.0.0
            exordis/application: cicd-sample
            exordis/application-instance: docs
            exordis/application-type: service
            exordis/environment: test
            exordis/product: Some Product
            exordis/subsystem: cicd
            helm.sh/chart: cicd-subsystem-application-0.1.0
        spec:
          containers:
            - envFrom:
                - configMapRef:
                    name: cicd-sample-docs-config-map-with-containers
                - configMapRef:
                    name: cicd-sample-docs-config-map-without-containers
                - configMapRef:
                    name: cicd-sample-docs-envs
                - secretRef:
                    name: cicd-sample-docs-gitlab-token
                - secretRef:
                    name: cicd-sample-docs-secret-with-containers
                - secretRef:
                    name: cicd-sample-docs-secret-without-containers
              image: registry.gitlab.com/cicd-unittests:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /healthcheck/live
                  port: 8080
                initialDelaySeconds: 10
                periodSeconds: 10
                timeoutSeconds: 10
              name: application
              readinessProbe:
                httpGet:
                  path: /healthcheck/ready
                  port: 8080
                initialDelaySeconds: 10
                periodSeconds: 10
                timeoutSeconds: 10
              resources:
                limits:
                  cpu: 200m
                  memory: 1024Mi
                requests:
                  cpu: 2m
                  memory: 70Mi
              startupProbe:
                failureThreshold: 20
                httpGet:
                  path: /healthcheck/live
                  port: 8080
                initialDelaySeconds: 10
                periodSeconds: 10
                timeoutSeconds: 10
              volumeMounts:
                - mountPath: /pvc-data
                  name: mypvc
                - mountPath: /data/templates/generated
                  name: templates-data-templates-generated
                - mountPath: /data/templates/original
                  name: templates-data-templates-original
                - mountPath: /some-data
                  name: volume1
            - envFrom:
                - configMapRef:
                    name: cicd-sample-docs-config-map-without-containers
                - configMapRef:
                    name: cicd-sample-docs-envs
                - secretRef:
                    name: cicd-sample-docs-gitlab-token
                - secretRef:
                    name: cicd-sample-docs-secret-without-containers
              image: registry.gitlab.com/jsreport:1.0.0
              imagePullPolicy: IfNotPresent
              name: jsreport
              resources:
                limits:
                  cpu: 200m
                  memory: 1024Mi
                requests:
                  cpu: 2m
                  memory: 70Mi
              volumeMounts:
                - mountPath: /jsreport
                  name: templates-data-templates-generated
                - mountPath: /some-another-data-but-same-as-for-applicationContainer
                  name: volume1
          initContainers:
            - envFrom:
                - configMapRef:
                    name: cicd-sample-docs-config-map-without-containers
                - configMapRef:
                    name: cicd-sample-docs-envs
                - secretRef:
                    name: cicd-sample-docs-gitlab-token
                - secretRef:
                    name: cicd-sample-docs-secret-without-containers
              image: registry.gitlab.com/my-default-image:1.0.0
              imagePullPolicy: IfNotPresent
              name: migration
              resources:
                limits:
                  cpu: 200m
                  memory: 1024Mi
                requests:
                  cpu: 2m
                  memory: 70Mi
              volumeMounts: null
            - envFrom:
                - configMapRef:
                    name: cicd-sample-docs-config-map-without-containers
                - configMapRef:
                    name: cicd-sample-docs-envs
                - secretRef:
                    name: cicd-sample-docs-gitlab-token
                - secretRef:
                    name: cicd-sample-docs-secret-without-containers
              image: registry.gitlab.com/software-development/subsystems/some-subsystem/data-content-templates:1.24.0-dev.5
              imagePullPolicy: Always
              name: templates
              resources:
                limits:
                  cpu: 200m
                  memory: 1024Mi
                requests:
                  cpu: 2m
                  memory: 70Mi
              volumeMounts:
                - mountPath: /data/templates/generated
                  name: templates-data-templates-generated
                - mountPath: /data/templates/original
                  name: templates-data-templates-original
          volumes:
            - name: mypvc
              persistentVolumeClaim:
                claimName: cicd-sample-docs-mypvc
            - emptyDir:
                sizeLimit: 200Mi
              name: templates-data-templates-generated
            - emptyDir:
                sizeLimit: 200Mi
              name: templates-data-templates-original
            - emptyDir:
                sizeLimit: 100Mi
              name: volume1
  7: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-gitlab-token
      namespace: cicd-test
    spec:
      data:
        - remoteRef:
            key: GITLAB_PYPI_REGISTRY_LOGIN
          secretKey: login
        - remoteRef:
            key: GITLAB_PYPI_REGISTRY_PASSWORD
          secretKey: password
      refreshInterval: 1m
      secretStoreRef:
        kind: ClusterSecretStore
        name: cicd-test
      target:
        creationPolicy: Owner
        deletionPolicy: Retain
        name: cicd-sample-docs-gitlab-token
        template:
          data:
            GITLAB_PYPI_REGISTRY_LOGIN: '{{ `{{ .login }}` }}'
            GITLAB_PYPI_REGISTRY_PASSWORD: '{{ `{{ .password }}` }}'
          engineVersion: v2
          metadata:
            annotations: null
            labels:
              app.kubernetes.io/component: cicd-sample
              app.kubernetes.io/instance: docs
              app.kubernetes.io/managed-by: helm
              app.kubernetes.io/name: cicd-sample
              app.kubernetes.io/part-of: cicd
              app.kubernetes.io/version: 1.0.0
              exordis/application: cicd-sample
              exordis/application-instance: docs
              exordis/application-type: service
              exordis/environment: test
              exordis/product: Some Product
              exordis/subsystem: cicd
              helm.sh/chart: cicd-subsystem-application-0.1.0
  8: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-mypvc
      namespace: cicd-test
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
  9: |
    apiVersion: v1
    kind: Secret
    metadata:
      annotations:
        extra-annotation: annotation value
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        extra-label: label value
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-secret-full-metadata
      namespace: my-namespace
    stringData:
      password: passw0rd
  10: |
    apiVersion: v1
    kind: Secret
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-secret-with-containers
      namespace: cicd-test
    stringData:
      password: passw0rd
  11: |
    apiVersion: v1
    kind: Secret
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: cicd-sample-docs-secret-without-containers
      namespace: cicd-test
    stringData:
      token: t0ken
  12: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        monitor-annotation: full service monitor
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
        monitor-labels: full service monitor
      name: cicd-sample-docs-full
      namespace: cicd-test
    spec:
      endpoints:
        - interval: 10s
          path: /extra-metrics
          port: api
        - interval: 30s
          port: metrics
          relabelings:
            - action: labeldrop
              regex: pod
      jobLabel: full-service
      namespaceSelector:
        matchNames:
          - cicd-test
      selector:
        matchLabels:
          exordis/application: cicd-sample
          exordis/application-instance: docs
          exordis/environment: test
          exordis/subsystem: cicd
  13: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        service-annotation: full service definition
      labels:
        app.kubernetes.io/component: cicd-sample
        app.kubernetes.io/instance: docs
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: cicd-sample
        app.kubernetes.io/part-of: cicd
        app.kubernetes.io/version: 1.0.0
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: cicd
        helm.sh/chart: cicd-subsystem-application-0.1.0
        service-label: full service definition
      name: cicd-sample-docs-full
      namespace: cicd-test
    spec:
      ports:
        - name: api
          port: 8080
          protocol: TCP
          targetPort: http
        - name: metrics
          port: 81
          protocol: TCP
          targetPort: 80
      selector:
        exordis/application: cicd-sample
        exordis/application-instance: docs
        exordis/environment: test
        exordis/subsystem: cicd
      sessionAffinity: None
      type: LoadBalancer
